app/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/NrsSubmissionJob.scala

```diff
@@ -17,33 +17,23 @@
 package uk.gov.hmrc.excisemovementcontrolsystemapi.scheduling
 
 import cats.implicits.toFunctorOps
+import org.apache.pekko.Done
 import play.api.{Configuration, Logging}
-import uk.gov.hmrc.excisemovementcontrolsystemapi.repository.NRSWorkItemRepository
 import uk.gov.hmrc.excisemovementcontrolsystemapi.services.NrsServiceNew
-import uk.gov.hmrc.excisemovementcontrolsystemapi.utils.DateTimeService
 import uk.gov.hmrc.http.HeaderCarrier
 
 import javax.inject.Inject
 import scala.concurrent.duration.FiniteDuration
 import scala.concurrent.{ExecutionContext, Future}
 
-class NrsSubmissionScheduler @Inject() (
-  nrsWorkItemRepository: NRSWorkItemRepository,
+class NrsSubmissionJob @Inject() (
   nrsService: NrsServiceNew,
-  configuration: Configuration,
-  dateTimeService: DateTimeService
+  configuration: Configuration
 ) extends ScheduledJob
     with Logging {
 
   override def execute(implicit ec: ExecutionContext): Future[ScheduledJob.Result] = {
-    val now                     = dateTimeService.timestamp()
-    val result: Future[Boolean] = nrsWorkItemRepository
-      .pullOutstanding(now, now)
-      .flatMap {
-        case None           => Future.successful(true)
-        case Some(workItem) =>
-          nrsService.submitNrs(workItem).map(_ => true)
-      }
+    val result: Future[Done] = nrsService.processAllWithLock()
```

Comment:
I don't love how this doesn't have a Failed state. It'd be worth knowing if this job/process actually fails.